test/jdk/java/util/Formatter/Basic.java

```diff
@@ -97,10 +86,11 @@ public static void main(String[] args) {
 
         BasicDateTime.test();
 
-        if (fail != 0)
-            throw new RuntimeException((fail + pass) + " tests: "
-                                       + fail + " failure(s), first", first);
-        else
-            out.println("all " + (fail + pass) + " tests passed");
+        if (fail != 0) {
+            throw new RuntimeException(String.format("%d tests: %d failure(s)" +
```

Comment:
You might consider including `", first"` with the rest of the message string, instead of concatenating it. That line might end up slightly long, but it may be worth it.

Also, use `%s` for`first`, as it's a `Throwable` ;)
(You could also perhaps change `first` -> `first.toString()` in the final argument to format, to clarify.)

Reply:
Good point, will reassign it to one line.

"Also, use %s forfirst, as it's a Throwable", 
I believe I am assigning %d to pass+fail and fail. I pass first as a Throwable to the runtime exception constructor.

Reply:
Oh, right. Carry on.

Reply:
Instead of using String.format() static method, using "...".formatted() is slightly shorter/concise.